test/comment-regex.js

```diff
@@ -51,15 +69,93 @@ test('comment regex new spec - #', function (t) {
     '\t/*# ', // multi line style with leading tab
     '/*# ',   // multi line style with leading text
   ].forEach(function (x) { 
-    t.ok(comment(x, ''), 'matches ' + x) 
-    t.ok(commentWithCharSet(x, ''), 'matches ' + x + ' with charset') 
-    t.ok(commentWithCharSet(x, '', '='), 'matches ' + x + ' with charset')
+    t.ok(comment(x, '', convert.commentRegex), 'matches ' + x)
+    t.ok(commentURI(x, '', convert.commentRegex), 'matches ' + x + ' uri')
+    t.ok(commentWithCharSet(x, '', convert.commentRegex), 'matches ' + x + ' with charset')
+    t.ok(commentURIWithCharSet(x, '', convert.commentRegex), 'matches ' + x + ' uri with charset')
+    t.ok(commentWithoutMediaType(x, '', convert.commentRegex), 'matches ' + x + ' without media type')
+    t.ok(commentURIWithoutMediaType(x, '', convert.commentRegex), 'matches ' + x + ' uri without media type')
+  });
+  
+  [ 
+    ' #// #',
+    ' #/* #',
+  ].forEach(function (x) {
+    t.ok(!comment(x, '', convert.commentRegex), 'should not match ' + x)
+    t.ok(!commentURI(x, '', convert.commentRegex), 'should not match ' + x + ' uri')
+  })
+
+  t.end()
+})
+
+test('comment regex groups', function (t) {
+  [ 
+    '  //# ', // with leading spaces
+    '\t//# ', // with leading tab
+    '//# ',   // with leading text
+    '/*# ',   // multi line style
+    '  /*# ', // multi line style with leading spaces
+    '\t/*# ', // multi line style with leading tab
+    '/*# ',   // multi line style with leading text
+  ].forEach(function (x) {
+    var m;
+    m = comment(x, '', convert.commentRegex)
+    t.ok(m, 'matches ' + x)
```

Comment:
It seems here we're accessing the matches so `rx.test` would not work, however we could have just added a custom method for this case and left the original ones alone

Reply:
I created a separate set of functions that use `exec` that only exist inside this test.